packages/slate/src/changes/at-range.js

```diff
@@ -28,7 +28,7 @@ const Changes = {}
 Changes.addMarkAtRange = (change, range, mark, options = {}) => {
   if (range.isCollapsed) return
 
-  const { normalize = true } = options
+  const normalize = change.getFlag('normalize', { normalize: true, ...options })
```

Comment:
I'm not sure if there's a better way to write this, but it is important that the logic is equivalent here. We need to only use this default value of `true` if `normalize` is not in `flags`  and not in `options`. It's a little difficult to express it concisely

Reply:
If the `Change` model's `constructor` set `this.flags.normalize = true` then I think we could avoid this?

Reply:
Sorry, I noticed that comment before and I forgot to address it! ;_;

The reason why that won't work in this existing implementation is that all `{ normalize: false }` option objects will fail because of the way `getFlag` operates. In the current implementation the assumption is that the `normalize` flag will override the `normalize` option, because that allows `withoutNormalization` to globally turn off normalization regardless of what callers pass in options

Does that make sense?

Reply:
@CameronAckermanSEL Ah, I think the `options` should be able to override the `flags`. And then `withoutNormalization` would just be a quick shortcut for changing the flags default value for all the change methods inside the callback. But if a user really wanted to (for some reason) override with `options` then that's fine—although strange.

Reply:
Alrighty, that's not a problem. I'll go rework that and hopefully have something for you to look at shortly :)

Reply:
ok! @ianstormtaylor got the desired behavior of options overriding flags, and adjusted tests to make sure it works properly. Tests should also now look at the case where someone unsets the normalize flag to make sure nothing blows up and the behavior for `withoutNormalization` still works correctly